src/Data/Rational/Unnormalised/Properties.agda

```diff
@@ -1670,9 +1678,27 @@ neg-distrib-⊓-⊔ = antimono-≤-distrib-⊓ neg-mono-≤
   ℤ.+ n ℤ.* 1ℤ       ∎)
   where open ℤ.≤-Reasoning
 
-∣p∣≡p∨∣p∣≡-p : ∀ p → (∣ p ∣ ≡ p) ⊎ (∣ p ∣ ≡ - p)
-∣p∣≡p∨∣p∣≡-p (mkℚᵘ (ℤ.+ n)    d-1) = inj₁ refl
-∣p∣≡p∨∣p∣≡-p (mkℚᵘ (-[1+ n ]) d-1) = inj₂ refl
+∣p∣≡p⊎∣p∣≡-p : ∀ p → (∣ p ∣ ≡ p) ⊎ (∣ p ∣ ≡ - p)
+∣p∣≡p⊎∣p∣≡-p (mkℚᵘ (ℤ.+ n)    d-1) = inj₁ refl
+∣p∣≡p⊎∣p∣≡-p (mkℚᵘ (-[1+ n ]) d-1) = inj₂ refl
+
+
+∣p∣≃p⇒0≤p : ∀ {p} → ∣ p ∣ ≃ p → 0ℚᵘ ≤ p
+∣p∣≃p⇒0≤p {p} ∣p∣≃p with ∣p∣≡p⊎∣p∣≡-p p
+∣p∣≃p⇒0≤p {p} ∣p∣≃p | inj₁ ∣p∣≡p  = ∣p∣≡p⇒0≤p ∣p∣≡p
+∣p∣≃p⇒0≤p {p} ∣p∣≃p | inj₂ ∣p∣≡-p =
+  ≤-reflexive (≃-sym (p+p≃0⇒p≃0 p eq))
+  where open ≤-Reasoning
```

Comment:
Any chance this could be laid out as:
```agda
≤-reflexive (≃-sym (p+p≃0⇒p≃0 p (begin-equality
  p + p      ≈⟨ +-congʳ p (≃-sym ∣p∣≃p) ⟩
  p + ∣ p ∣  ≈⟨ +-congʳ p (≃-reflexive ∣p∣≡-p) ⟩
  p - p      ≈⟨ +-inverseʳ p ⟩
  0ℚᵘ        ∎)))
  where open ≤-Reasoning
```

Though I suspect this case could be made simpler and more reusable by adding another simple lemma `p≃-p⇒p≃0` and then this case would become simply `rewrite ∣p∣≡-p = ≤-reflexive (p≃-p⇒p≃0 ∣p∣≃p)`

Reply:
I made the lemma as you suggested, and altered the proof. It wasn't quite as nice due to the need for some uses of symmetry, but it is a bit better.